e2e/ibc_test.go

```diff
@@ -0,0 +1,207 @@
+	sourceManagerAddress = common.FromHex("0x00000000000000000000000083ae82bd4054e815fb7b189c39d9ce670369ea16")
+	// recipientManagerAddress is the 32-byte representation of the "dollar/manager" module account.
+	recipientManagerAddress = common.FromHex("0x0000000000000000000000002e859506ba229c183f8985d54fe7210923fb9bca")
+
+	// channelID is the IBC channel identifier of ibc-go-simd.
+	channelID = "channel-0"
+	// yieldRecipient is the "gov" module account on ibc-go-simd.
+	yieldRecipient = "cosmos10d07y265gmmuvt4z0w9aw880jnsr700j6zn9kn"
+)
+
+// TestIBCYieldDistribution tests $USDN yield distribution across IBC channels.
+func TestIBCYieldDistribution(t *testing.T) {
+	ctx, chain, externalChain, guardians := Suite(t, true)
+	validator := chain.Validators[0]
+
+	// ARRANGE: Recover the authority wallet in order to perform gated actions.
+	// This mnemonic must be for the authority account set in simapp/app.yaml!
+	authority, err := interchaintest.GetAndFundTestUserWithMnemonic(ctx, "authority", "occur subway woman achieve deputy rapid museum point usual appear oil blue rate title claw debate flag gallery level object baby winner erase carbon", math.OneInt(), chain)
+	require.NoError(t, err)
+
+	// ARRANGE: Create and fund test user accounts on both Noble and the external chain.
+	wallets := interchaintest.GetAndFundTestUsers(t, ctx, "user", math.OneInt(), chain, externalChain)
+	user, externalUser := wallets[0], wallets[1]
+
+	// ARRANGE: Pad the user address to be compatible with Wormhole's NTT standard.
+	paddedAddress := make([]byte, 32)
+	copy(paddedAddress[12:], user.Address())
+
+	// ARRANGE: Prepare a VAA to be delivered that mints the user 1,000,000 $USDN.
+	additionalPayload := portaltypes.EncodeAdditionalPayload(1e12, destinationToken)
+	payload := ntt.EncodeNativeTokenTransfer(ntt.NativeTokenTransfer{
+		Amount:            1_000_000 * 1e6,
+		SourceToken:       sourceToken,
+		To:                paddedAddress,
+		ToChain:           uint16(vaautils.ChainIDNoble),
+		AdditionalPayload: additionalPayload,
+	})
+
+	transceiverMessage := buildTransceiverMessage(payload)
+	vaa := utils.NewVAA(guardians, transceiverMessage)
+
+	bz, err := vaa.Marshal()
+	require.NoError(t, err)
+
+	// ACT: Deliver the prepared VAA that mints 1,000,000 $USDN.
+	_, err = validator.ExecTx(
+		ctx, authority.KeyName(),
+		"dollar", "portal", "deliver", base64.StdEncoding.EncodeToString(bz),
+	)
+	require.NoError(t, err)
+
+	// ASSERT: The user should now have 1,000,000 $USDN.
+	balance, err := chain.BankQueryBalance(ctx, user.FormattedAddress(), "uusdn")
+	require.NoError(t, err)
+	require.Equal(t, math.NewInt(1_000_000*1e6), balance)
+
+	// ACT: Query all yield recipients.
+	yieldRecipients := getYieldRecipients(t, ctx, validator)
+
+	// ASSERT: There are no yield recipients.
+	require.Empty(t, yieldRecipients)
+
+	// TODO: Once NobleICS4Wrapper has been migrated, ensure that transfers are blocked at first!
```

Comment:
Ahh i asked about a similar comment elsewhere in my review but I think I understand what you mean now.

We need to block $USDN transfers until a yield recipient has been registered for that channel.